mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp

```diff
@@ -1171,28 +1171,30 @@ static FuncOp createVerifyFuncOp(ModuleOp &module, OpBuilder &builder,
 
   scf::IfOp ifOp;
   if (outputMemRefType.getElementType() == builder.getIntegerType(16)) {
-    auto cmpOp = bt4.create<CmpIOp>(builder.getUnknownLoc(), CmpIPredicate::ne,
+    auto cmpOp = bt4.create<arith::CmpIOp>(builder.getUnknownLoc(), arith::CmpIPredicate::ne,
                                     cpuLoadOp, gpuLoadOp);
     ifOp = bt4.create<scf::IfOp>(builder.getUnknownLoc(), cmpOp, false);
   } else {
     if (randomSeed.getValue() != "none" &&
         randomDataType.getValue() == "float") {
-      float delta = 0.0000001;
+      std::string delta = "0.0000001";
       if (outputMemRefType.getElementType() == builder.getF16Type())
-        delta = 0.0001;
+        delta = "0.0001";
 
-      auto deltaConstantOp = bt4.create<ConstantOp>(
-          builder.getUnknownLoc(), outputMemRefType.getElementType(),
-          builder.getFloatAttr(outputMemRefType.getElementType(), delta));
+      auto deltaType = outputMemRefType.getElementType().cast<FloatType>();
+      auto deltaConstantOp = bt4.create<arith::ConstantFloatOp>(
+          builder.getUnknownLoc(),
+          APFloat(deltaType.getFloatSemantics(), delta),
```

Comment:
There was a runtime complaint about trying to make an F16 op with default F32 semantics, plus making a ConstantFloatOp was a little more constrained than making a ConstantOp with a FloatAttr.  This form at least matches up the semantics and type.